modules/silverpushBidAdapter.js

```diff
@@ -0,0 +1,326 @@
+
+const VIDEO_ORTB_PARAMS = [
+  'mimes',
+  'minduration',
+  'maxduration',
+  'placement',
+  'protocols',
+  'startdelay',
+  'skip',
+  'skipafter',
+  'minbitrate',
+  'maxbitrate',
+  'delivery',
+  'playbackmethod',
+  'api',
+  'linearity'
+];
+
+export const VIDEO_ORTB_REQUIRED = ['api', 'mimes', 'placement', 'protocols', 'minduration', 'maxduration', 'startdelay'];
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO],
+  isBidRequestValid,
+  buildRequests,
+  interpretResponse,
+  onBidWon,
+  getRequest: function(endpoint) {
+    ajax(endpoint, null, undefined, {method: 'GET'});
+  },
+  getOS: function(ua) {
+    if (ua.indexOf('Windows') != -1) { return 'Windows'; } else if (ua.match(/(iPhone|iPod|iPad)/)) { return 'iOS'; } else if (ua.indexOf('Mac OS X') != -1) { return 'macOS'; } else if (ua.match(/Android/)) { return 'Android'; } else if (ua.indexOf('Linux') != -1) { return 'Linux'; } else { return 'Unknown'; }
+  }
+};
+
+registerBidder(spec);
+
+export const CONVERTER = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  imp(buildImp, bidRequest, context) {
+    let imp = buildImp(bidRequest, context);
+
+    if (bidRequest.mediaTypes[VIDEO]) {
+      imp = buildVideoImp(bidRequest, imp);
+    } else if (bidRequest.mediaTypes[BANNER]) {
+      imp = buildBannerImp(bidRequest, imp);
+    }
+
+    const bidFloor = getBidFloor(bidRequest);
+
+    utils.deepSetValue(imp, 'bidfloor', bidFloor);
+
+    if (bidRequest.params.deals && bidRequest.params.deals.length > 0) {
+      utils.deepSetValue(imp, 'pmp', { deals: bidRequest.params.deals });
+    }
+
+    return imp;
+  },
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+      ext: {
+        bc: `${bidderConfig}_${bidderVersion}`
+      }
+    })
+
+    let userAgent = navigator.userAgent;
+    utils.deepSetValue(req, 'device.os', spec.getOS(userAgent));
+    utils.deepSetValue(req, 'device.devicetype', _isMobile() ? 1 : _isConnectedTV() ? 3 : 2);
+
+    const bid = context.bidRequests[0];
+    if (bid.params.publisherId) {
+      utils.deepSetValue(req, 'site.publisher.id', bid.params.publisherId);
+    }
+
+    return req;
+  },
+
+  bidResponse(buildBidResponse, bid, context) {
+    let bidResponse = buildBidResponse(bid, context);
+
+    if (bid.ext) {
+      bidResponse.meta.networkId = bid.ext.dsp_id;
+      bidResponse.meta.advertiserId = bid.ext.buyer_id;
+      bidResponse.meta.brandId = bid.ext.brand_id;
+    }
+
+    if (context.ortbResponse.ext && context.ortbResponse.ext.paf) {
+      bidResponse.meta.paf = Object.assign({}, context.ortbResponse.ext.paf);
+      bidResponse.meta.paf.content_id = utils.deepAccess(bid, 'ext.paf.content_id');
+    }
+
+    bidResponse = buildVideoVastRespose(bidResponse)
```

Comment:
typo, should be `buildVideoVastResponse`

Reply:
Typo fixed.